### PR TITLE
FIX: Set UseHighDpiPixmaps only for PyQt5 and PySide2

### PR DESCRIPTION
**Pull request by florin-pop**
#### Background

In https://github.com/mne-tools/mne-nirs/pull/510 we migrated to `PySide6`.
When calling [`setAttribute(Qt.AA_UseHighDpiPixmaps)`](https://github.com/mne-tools/mne-python/blob/cecbf0fb1dc919254bb6095db491b2d7c5f68003/mne/viz/backends/_utils.py#L164) there is a warning as this attribute was deprecated in [QT6](https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/global/qnamespace.h?h=dev#n437) and it has no effect as HiDPI is enabled by default.

#### What does this implement/fix?

One way of solving the issue, as suggested by @larsoner in https://github.com/mne-tools/mne-nirs/pull/510#discussion_r1179883423 is to check the QT version and set the attribute conditionally.

---
Contributed with ❤️ by AE Studio

### CODE REVIEW

File: mne/utils/check.py
Comment (florin-pop) [inline mne/utils/check.py:862]: I added a new argument to prevent `_check_qt_version` and `_init_mne_qtapp` from calling eachother endlessly. Do you see a better approach?

File: mne/viz/backends/_utils.py
Comment (florin-pop) [inline mne/viz/backends/_utils.py:168]: I don't think that the `try` - `except` is needed anymore, but I wanted to be conservative in this change. Let me know if you think that I should remove it.

Comment (larsoner) [reply mne/viz/backends/_utils.py:168]: Agreed you can remove it

File: mne/utils/check.py
Comment (larsoner) [reply mne/utils/check.py:862]: I think this is good!

**Review by larsoner — COMMENTED** [review]
Can you just add a quick line to `doc/changes/latest.inc` with `:newcontrib:` to the top of the BUG section about improving compat with Qt6 and add your name to `doc/changes/names.inc`? Then after removing the `try` I think we can merge!

**Comment by florin-pop** [comment]
Thank you @larsoner!

**Review by larsoner — APPROVED** [review]
Will merge when green, thanks @florin-pop !

File: doc/changes/latest.inc
Comment (larsoner) [inline doc/changes/latest.inc:51]:
```suggestion
- Improving compatibility with Qt6 by removing the use of deprecated ``AA_UseHighDpiPixmaps`` attribute for this Qt version (:gh:`11662` by :newcontrib:`Florin Pop`)
```

**Comment by rob-luke** [comment]
🎉 newcontrib 🎉

**Pull request merged**
